services/thermal_mgr/thermal_mgr.c

```diff
@@ -24,48 +24,95 @@ static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR
 
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
+  error_code_t errCode;
+  errCode = xQueueSend(thermalMgrQueueHandle, (void *)event, 10);
+  if (errCode != ERR_CODE_SUCCESS)
+    return errCode;
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
-  /* Implement this function */
+void osHandlerLM75BD(void)
+{
+  thermal_mgr_event_t overTemp;
+  overTemp.type = THERMAL_MGR_EVENT_CHECK_TEMP_CMD;
+  thermalMgrSendEvent(&overTemp);
 }
 
-static void thermalMgr(void *pvParameters) {
+static void thermalMgr(void *pvParameters)
+{
+  if (pvParameters == NULL) {return;}
+
   /* Implement this task */
-  while (1) {
-    
+  while (1)
+  {
+    if (thermalMgrQueueHandle != NULL)
+    {
+        thermal_mgr_event_t event;
+      if (xQueueReceive(thermalMgrQueueHandle, &event, 10) == pdPASS)
```

Comment:
This task doesn't do anything unless there's an event in the queue. So just block indefinitely